### PR TITLE
Add support for composable roles for OSP17

### PR DESCRIPTION
**Pull request by asyedham**

### CODE REVIEW

**Comment by tosky** [comment]
Out of curiosity, have you tried using the new generate_baremetal_deployment_from_roles_data.py script which is used by default by infrared in the composable role case, and which may help reducing some duplication (by relying on some files provided by THT)?

**Comment by asyedham** [comment]
@tosky That's interesting, we haven't used this  new generate_baremetal_deployment_from_roles_data.py script, is there any example/ doc how this creates the composable roles as in our  case the role is create based on hardware type like Compute1029u,Computer630, Compute1029p so on

File: composable_prepare_nic_configs.yml [hunk 1]
```diff
@@ -243,3 +263,60 @@
...
+            delegate_to: undercloud
+            with_together:
+              - "{{ machine_types }}"
+              - "{{ ctlplane_interfaces }}"
+              - "{{ external_interfaces }}"
+              - "{{ tenant_interfaces }}"
+              - "{{ isolated_interfaces }}"
+              - "{{ openshift_vlan_interfaces }}"
+
+        when: osp_release|int == 17
```
Comment (masco) [inline composable_prepare_nic_configs.yml]: can you use >= 17, it won't work on the later versions. it will be good to change all the == 17 to >= 17

**Review by masco — CHANGES_REQUESTED** [review]

File: composable_prepare_nic_configs.yml (hunk 1 above)
Comment (asyedham) [reply composable_prepare_nic_configs.yml]: Done

Comment (masco) [reply composable_prepare_nic_configs.yml]: still i am seeing ==

**Pull request merged**